### ISSUE TITLE
pyOpenMS builds broken

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#5188]
potentially related to https://github.com/OpenMS/OpenMS/pull/4922

**Comment by jpfeuffer** [comment]
If you look into the nightly pyoms wheels action, you see that it works with py3.6 but fails with 3.7+.
Explains why 3.6 on travis went through in the PR.

**Comment by oliveralka** [comment]
Can confirm, that everything works with python 3.6 and autowrap 0.22.1. 
Issues arise using python 3.7 upwards. 

Compiling with python 3.9 throws the following error:
```
Traceback (most recent call last):
  File "/Users/alka/Documents/work/software/openms_pyopenms/20210224_test/pyOpenMS/create_cpp_extension.py", line 208, in <module>
    autowrap_include_dirs = doCythonCodeGeneration(modname, allDecl_mapping, instance_map, converters)
  File "/Users/alka/Documents/work/software/openms_pyopenms/20210224_test/pyOpenMS/create_cpp_extension.py", line 168, in doCythonCodeGeneration
    autowrap_include_dirs = autowrap.generate_code(allDecl_mapping[modname]["decls"], instance_map,
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/__init__.py", line 66, in generate_code
    gen.create_pyx_file(debug)
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 216, in create_pyx_file
    create_for(ResolvedClass, self.create_wrapper_for_class)
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 212, in create_for
    method(resolved)
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 472, in create_wrapper_for_class
    codes = self.create_wrapper_for_method(r_class, name, methods)
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 635, in create_wrapper_for_method
    code = self.create_wrapper_for_nonoverloaded_method(cdcl,
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 786, in create_wrapper_for_nonoverloaded_method
    call_args, cleanups, in_types = self._create_fun_decl_and_input_conversion(meth_code,
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/CodeGenerator.py", line 674, in _create_fun_decl_and_input_conversion
    checks.append((n, converter.type_check_expression(t, n)))
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/ConversionProvider.py", line 1056, in type_check_expression
    inner_check = inner_conv.type_check_expression(tt, arg_var_next)
  File "/usr/local/miniconda3/envs/build_pyopenms_39/lib/python3.9/site-packages/autowrap/ConversionProvider.py", line 243, in type_check_expression
    raise NotImplementedError("void has no matching python type")
NotImplementedError: void has no matching python type
```

Which seems to point to an autowrap issue. 

Using some rudimentary debugging (print statements) in autowraps `ConversionProvider.py` and `CodeGenerator.py` point to:

```
void PeakIntensityPredictor()
_predict_0
double predict(AASequence & sequence)
_predict_1
double predict(AASequence & sequence, libcpp_vector[double] & add_info)
_predict_2
libcpp_vector[double] predict(libcpp_vector[AASequence] & sequences)
_predict_3
libcpp_vector[double] predict(libcpp_vector[AASequence] & sequences, libcpp_vector[void] & add_info)
<autowrap.ConversionProvider.VoidConverter object at 0x7ffaa59df3a0>
```

https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/pxds/PeakIntensityPredictor.pxd
This could indicate issues with nestes structures such as `libcpp_vector[ libcpp_vector[ double ] ] & add_info`

But I have to admit that it is not clear to me what that would have to do with the python version issue. 
My best guess is, that this is only a symptom of an entirely different issue.

**Comment by timosachsenberg** [comment]
I think it could be the void in libcpp_vector[void] ?

**Comment by jpfeuffer** [comment]
Yes, it seems like we parse the second level of the template as void.

I remembered that we actually import the Cython parser to parse the pxds, so we might need to double-check the cython versions as well.

**Comment by jpfeuffer** [comment]
Yes, a difference between the last successful build and the current failing ones is also the jump from Cython 0.29.21 to 0.29.22.
Interestingly 0.29.22 included a "fix" for nested templates:
https://github.com/cython/cython/pull/3886

**Comment by oliveralka** [comment]
Yes, @jpfeuffer you are correct. 
I did a local test build with Python 3.9 and Cython 0.29.21 and everything proceeds normally.

**Comment by jpfeuffer** [comment]
By the way, it could be just our style of putting spaces between the brackets.

**Comment by oliveralka** [comment]
I tried to remove the spaces between the brackets in the `PeakIntensityPredictor` Example, but it does not seem to have an effect.

**Comment by oliveralka** [comment]
It works locally with python 3.9, cython 0.29.22 and autowrap 0.22.2.
See https://github.com/uweschmitt/autowrap/issues/99

**Comment by timosachsenberg** [comment]
k great!

**Comment by jpfeuffer** [comment]
Fixed in #5193
